### PR TITLE
Bring back UCX 1.14

### PR DESCRIPTION
**Pull request by abellina**
Reverts (the revert): https://github.com/NVIDIA/spark-rapids/pull/7942

This updates the pom, dockerfile, and documentation to UCX 1.14.

Note that the Dockerfile that we have is an example for Ubuntu 20.04. Ubuntu 22.04 looks to have other issues with the linkage of the rdma-core I had been compiling int he dockerfile. I am looking for confirmation on where to go next for that example, but it should be a documentation change later on how to install things for Ubuntu 22.04 only.

### CODE REVIEW

**Comment by abellina** [comment]
The CI issues we have seen here https://github.com/NVIDIA/spark-rapids/issues/7940 are related to pre-existing TCP keepalive issues in UCX.

**Comment by abellina** [comment]
build

File: docs/additional-functionality/rapids-shuffle.md [hunk 1]
```diff
@@ -126,52 +118,53 @@ The required UCX version for the RAPIDS Shuffle Manager is
...
 
-##### CentOS UCX RPM
-The UCX packages for CentOS 7 and 8 are divided into different RPMs. For example, UCX 1.12.1
-available at
-https://github.com/openucx/ucx/releases/download/v1.12.1/ucx-v1.12.1-centos7-mofed5-cuda11.tar.bz2
+##### Rocky UCX RPM
+The UCX packages for Rocky are divided into different RPMs.
+
+For example, UCX 1.14.0 available at
+https://github.com/openucx/ucx/releases/download/v1.14.0/ucx-v1.14.0-centos8-mofed5-cuda11.tar.bz2
```
Comment (sameerz) [inline docs/additional-functionality/rapids-shuffle.md:129]:
```suggestion
https://github.com/openucx/ucx/releases/download/v1.14.0/ucx-1.14.0-centos8-mofed5-cuda11.tar.bz2
```

**Review by sameerz — CHANGES_REQUESTED** [review]

**Comment by abellina** [comment]
build

**Comment by abellina** [comment]
@sameerz let me know if this looks good. CI is building which will test fetching the jucx-1.14 jar and running a smoke test with it.

**Comment by abellina** [comment]
@jlowe @revans2 @tgravescs if I could get some :eyes: from you that would be great.

File: docs/additional-functionality/rapids-shuffle.md [hunk 2]
```diff
@@ -55,9 +55,8 @@ pools is the number of cores in the system divided by the number of executors pe
 ---
 **NOTE:**
 
-As of the spark-rapids 23.02 release, Ubuntu 22.04 UCX packages are not available. They
-will be available for future releases.
-
+As of the spark-rapids 23.06 release, UCX packages support CUDA 11.
+UCX support for CUDA 12 in RAPIDS Spark will be added in a future release.
```
Comment (jlowe) [inline docs/additional-functionality/rapids-shuffle.md:59]:
```suggestion
UCX support for CUDA 12 in the RAPIDS Accelerator will be added in a future release.
```

**Review by revans2 — DISMISSED** [review]
I did not actually test out the Dockerfiles, but I assume that someone did. If not please lat me know so I can revert my approval.

**Review by jlowe — APPROVED** [review]

**Comment by abellina** [comment]
> I did not actually test out the Dockerfiles, but I assume that someone did. If not please lat me know so I can revert my approval.

I thought I re-ran this, but I obviously I missed something. I re-build all of them and found an issue, unfortunately, with Ubuntu 22.04 and the rdma-core that this builds and installs. I also found an inconsistency in the no-rdma version of the ubuntu example. Fixed both https://github.com/NVIDIA/spark-rapids/pull/8402/commits/0e22f2502629b4774a01001aa991a0f503baa4da. 

This are example dockerfiles. I'll need to edit these files for Ubuntu 22.04, but I'll need to do that as a separate PR and we can add an example for Ubuntu 22.04 specifically or make it obvious what the differences are.

**Comment by abellina** [comment]
Found one more runtime issue with the dockerfile. I'll push shortly.

**Review by jlowe — APPROVED** [review]

**Comment by abellina** [comment]
build

**Review by sameerz — APPROVED** [review]

**Pull request merged**